acr purge operation encounter invalid memory error

There is a scheduled acr run task to purge images older than 90 days from one acr (viennadroptest.azurecr.io) by using purge container (_mcr.microsoft.com/acr/acr-cli:0.1 purge_). it often encountered invalid memory error (panic: runtime error: invalid memory address or nil pointer dereference) at worker code line [here](https://github.com/Azure/acr-cli/blob/master/cmd/worker/worker.go#L69)

```
Skipped viennadroptest.azurecr.io/data-store:2019101995651, HTTP status: 404
Skipped viennadroptest.azurecr.io/data-store:2019101718307, HTTP status: 404
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0xa4ec7e]

goroutine 70307 [running]:
github.com/Azure/acr-cli/cmd/worker.(*PurgeWorker).ProcessJob.func1(0xc0002bf5c0, 0xc000436120, 0x19, 0x7ffd1536676f, 0xa, 0xc004d90c12, 0xe, 0xc0055e4550, 0x47, 0x17a79d97, ...)
        /go/src/github.com/Azure/acr-cli/cmd/worker/worker.go:69 +0x4de
created by github.com/Azure/acr-cli/cmd/worker.(*PurgeWorker).ProcessJob
        /go/src/github.com/Azure/acr-cli/cmd/worker/worker.go:61 +0xa1
2020/03/14 21:36:36 Container failed during run: acb_step_0. No retries remaining.
failed to run step ID: acb_step_0: exit status 2

Run ID: cfs0 failed after 3h18m30s. Error: failed during run, err: exit status 1
```


![image](https://user-images.githubusercontent.com/11655855/76693770-15e0cb00-6628-11ea-817c-cec436867ad7.png)

**Additional context**
About 270 repositories in the acr occupied around 9T bytes storage

Thanks for replying my question. Basically I create acr run task

**Setup purge command**
PURGE_CMD="mcr.microsoft.com/acr/acr-cli:0.1 purge \
  --registry myregistry \
  --filter 'account-rp:.*' \
  --filter 'activator-ecd51ca5034883acbe737fde417a3d86:.*' \
  --filter 'aether/adlcloud:.*' \
  ...... ( about 270 filiters here
  --untagged \
  --ago 90d"

**schedule daily task**
az acr task create \
  --name dailyPurgeTask \
  --registry myregistry \
  --cmd "$PURGE_CMD" \
  --subscription <mysubscriptionid>\
  --timeout 28800 \
  --context /dev/null \
  --schedule "dailyTimer:0 3 * * Mon-Fri"

Then scheduled run randomly happened invliad memory issue. 

**Here the most recent error log**

Skipped viennadroptest.azurecr.io/artifact:2019092118275, HTTP status: 404
Skipped viennadroptest.azurecr.io/artifact:20190922191131, HTTP status: 404
Skipped viennadroptest.azurecr.io/artifact:20190921154815, HTTP status: 404
Skipped viennadroptest.azurecr.io/artifact:20190923132615, HTTP status: 404
Skipped viennadroptest.azurecr.io/artifact:20190924115948, HTTP status: 404

Skipped viennadroptest.azurecr.io/artifact:20190924123646, HTTP status: 404
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0xa4ec7e]

goroutine 77840 [running]:
github.com/Azure/acr-cli/cmd/worker.(*PurgeWorker).ProcessJob.func1(0xc000316450, 0xc000042940, 0x19, 0x7fffc5783625, 0x8, 0xc0039b3650, 0xd, 0xc0030ed8b0, 0x47, 0x101b466b, ...)
        /go/src/github.com/Azure/acr-cli/cmd/worker/worker.go:69 +0x4de
created by github.com/Azure/acr-cli/cmd/worker.(*PurgeWorker).ProcessJob
        /go/src/github.com/Azure/acr-cli/cmd/worker/worker.go:61 +0xa1
2020/03/23 06:33:05 Container failed during run: acb_step_0. No retries remaining.
failed to run step ID: acb_step_0: exit status 2

Run ID: cfsb failed after 3h32m9s. Error: failed during run, err: exit status 1

--- a/cmd/worker/worker.go
+++ b/cmd/worker/worker.go
@@ -66,7 +66,7 @@ func (pw *PurgeWorker) ProcessJob(ctx context.Context, job PurgeJob) {
 			// In case a tag is going to be purged DeleteAcrTag method is used.
 			resp, err := pw.acrClient.DeleteAcrTag(ctx, job.RepoName, job.Tag)
 			if err != nil {
-				if resp != nil && resp.StatusCode == http.StatusNotFound {
+				if resp != nil && resp.Response != nil && resp.StatusCode == http.StatusNotFound {
 					// If the tag is not found it can be assumed to have been deleted.
 					fmt.Printf("Skipped %s/%s:%s, HTTP status: %d\n", job.LoginURL, job.RepoName, job.Tag, resp.StatusCode)
 				} else {
@@ -82,7 +82,7 @@ func (pw *PurgeWorker) ProcessJob(ctx context.Context, job PurgeJob) {
 			// In case a manifest is going to be purged DeleteManifest method is used.
 			resp, err := pw.acrClient.DeleteManifest(ctx, job.RepoName, job.Digest)
 			if err != nil {
-				if resp != nil && resp.StatusCode == http.StatusNotFound {
+				if resp != nil && resp.Response != nil && resp.StatusCode == http.StatusNotFound {
 					// If the manifest is not found it can be assumed to have been deleted.
 					fmt.Printf("Skipped %s/%s@%s, HTTP status: %d\n", job.LoginURL, job.RepoName, job.Digest, resp.StatusCode)
 				} else {

